lib/doubly_linked_list.rb

```diff
@@ -0,0 +1,46 @@
+# doubly_linked_list.rb
+class Node 
+    attr_reader :data # allow external entities to read value but not write
+    attr_accessor :next # allow external entities to read or write next node
+    attr_accessor :previous # allow external entities to read or write previous node
+
+    def initialize(value, next_node = nil, previous_node = nil)
+      @data = value
+      @next = next_node
+      @previous = previous_node
+    end
+end
+
+class DoublyLinkedList 
+
+    def initialize
+        @head = nil
```

Comment:
With a doubly linked list I suggest having an `@tail` instance variable as it makes operations on the back of a Doubly linked list much easier.